### ISSUE TITLE
Vertical lines drawn with draw.line have one extra width pixel

### ISSUE THREAD

**Issue by Terseus** [python-pillow/Pillow#367]
When using the _line_ method of a _Draw_ object on a newly created image to draw vertical lines, all lines drawn with more than 1 pixel width are one pixel widther (a 2 pixels width line is drawn with a 3 pixels width).
##### How to reproduce the problem:

This "proof of concept" program reproduces the problem creating a 10x10 image with a centered cross of two lines 2 pixels width, one horizontal and one vertical, where the vertical one is clearly widther than the horizontal.

``` python
from PIL import Image, ImageDraw

img = Image.new('RGB', (10, 10), (255, 255, 255))
draw = ImageDraw.Draw(img)
draw.line((0, 5, 10, 5), (0, 0, 0), 2)
draw.line((5, 0, 5, 10), (0, 0, 0), 2)
img.save("test-bad-lines.png", 'PNG')
```
##### Expected result:

A cross of two lines, both with the same width, 2 pixels.
##### Current result:

A cross with the horizontal line 2 pixels width and the vertical line 3 pixels width.
##### Additional information:

This bug only happens when drawing **vertical** lines with **at least** two pixels width.
1 pixel width vertical lines and horizontal lines of any width doesn't seem to have this problem.

**Comment by benjimin** [comment]
The PIL documentation explicitly asserts that the coordinate origin is at the *top left corner* (not the centre) of the top left pixel. (E.g. see "pillow | docs | handbook | concepts | coordinate system".) Thus a square 10 pixel by 10 pixel image canvas spans from (0,0) to (10,10) in coordinate space; (10,10) is the coordinate for the bottom-right corner of the last pixel. 

Therefore, in Terseus example, the polygon [(2,2), (2,8), (8,8), (8,2)] **should be a square with sides of length 6 pixels**. *Not seven*, because that would encroach (overlap) on the polygon [(8,8), (8,10), (10,10), (10,8)], which clearly does not overlap in coordinate space (and is already perfectly aligned with the edges of the pixels). Trying to "include endpoints" can't make consistent sense (since in geometry, it is arbitrary which side you call the start versus the end).

**Comment by radarhere** [comment]
https://github.com/python-pillow/Pillow/issues/367#issuecomment-39967895
> So rectangle() explicitly says the "second point is just outside the drawn rectangle".

This was changed with #6625 to say that "The bounding box is inclusive of both endpoints."

**Comment by radarhere** [comment]
@Lonami looks like the original problem here was resolved by #737

The rest of this is a debate about whether shapes should include their endpoints or not.
